x-pack/plugins/canvas/public/lib/filter_adapters.ts

```diff
@@ -36,7 +36,7 @@ const collectArgs = (args: ExpressionFunctionAST['arguments']) => {
   );
 };
 
-export function adaptCanvasFilter(filter: ExpressionFunctionAST): Filter {
+export function adaptCanvasFilter(filter: AstFunction): Filter {
```

Comment:
nit: i preferred the old names `ExpressionFunctionAST` over just `AstFunction`

Reply:
I had a few points to rename them.
1. They were inconsistent `Ast` versus`ExpressionFunctionAST` and `ExpressionArgAST` (`Ast` vs `AST`, and not `ExpressionAST` for some reason).
2. `ExpressionArgAST` - shortcut in the name is not a good practice.
3. The `AST` context was in the suffix, but it should be in the prefix.
4. The `AST` was incorrectly camel-cased and should be `Ast`.
5. It was confusing next to the expressions plugin's types (`ExpressionFunctionAST` & `ExpressionAstFunction`, `ExpressionArgAST` & `ExpressionAstArgument`).

I think they are named better know:
- Clear `Ast` and `ExpressionAst` prefixes.
- Consistent with each other and with the expressions plugin types.

WDYT?